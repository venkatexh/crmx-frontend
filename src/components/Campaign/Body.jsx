import React from "react";
import "../../sass/components/contact/contact.component.scss";

const Body = ({ campaign }) => {
  const { subject, text, tags } = campaign;
  
  return (
    <div className={"contactBody"}>
      <div className={"left"}>
        <div className={"heading"}>Email Content</div>
        <div>Subject</div>
        <div>{subject}</div>
        <div>Email Text</div>
        <div>{text}</div>
      </div>
      <div className={"right"}>
        <div className={"heading"}>Tags</div>
        <div className={"tagsList"}>
          {tags?.map((tag, idx) => {
            return (
              <div className={"contactTag"} key={idx}>
                {tag.name}
              </div>
            );
          })}
        </div>
      </div>
    </div>
  );
};

export default React.memo(Body);
