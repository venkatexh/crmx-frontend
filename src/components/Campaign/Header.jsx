import "../../sass/components/contact/contact.component.scss";

const Header = ({campaign}) => {
  const {name, from, status} = campaign;
  return <div className={'contactHeader'}>
    <div className={'left'}>
      <div className={'name'}>{name}</div>
      <div>From {from}</div>
    </div>
    <div>
      <div>Added 2nd Jan 2022</div>
      <div>Last Updated 2nd Jan 2022</div>
    </div>
    <div>Actions</div>
    <div>{status}</div>
  </div>
}

export default Header;