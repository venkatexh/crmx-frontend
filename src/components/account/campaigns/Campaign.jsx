import "../../../sass/components/account/campaigns/campaign.component.scss";
const colors = {
  draft: "#e8eded",
  paused: "#e8eded",
  sent: "#6ef589",
  scheduled: "#ebe660",
  failed: "#d6063a",
};

const Campaign = ({ name, status, scheduledAt, sentAt, sentTo }) => {
  return (
    <div className={"campaign"}>
      <div className={"top"}>
        <div className={"name"}>{name}</div>
        <div className={"status"} style={{ background: colors[status] }}>
          {status}
        </div>
      </div>
      <div className={"bottom"}>
        <div>
          <div>
            {status === "sent"
              ? `Sent ${sentAt}`
              : status === "scheduled"
              ? `Scheduled at ${scheduledAt}`
              : ""}
          </div>
          <div>{sentTo.length} recipients</div>
        </div>
        <div>
          <button className={"button"}>View Campaign</button>
        </div>
      </div>
    </div>
  );
};

export default Campaign;
