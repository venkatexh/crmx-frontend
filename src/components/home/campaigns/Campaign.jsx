import "../../../sass/components/home/campaigns/campaign.component.scss";
import { Link } from "react-router-dom";
import Axios from "axios";
import hostHeader from "../../../config/host";
import { memo, useState } from "react";

const colors = {
  draft: "#e8eded",
  paused: "#e8eded",
  Sent: "#6ef589",
  Scheduled: "#ebe660",
  failed: "#d6063a",
};

const Campaign = ({ name, status, scheduledAt, sentAt, sentTo, id }) => {
  const [success, setSuccess] = useState(false);
  const [error, setError] = useState(false);
  const [loading, setLoading] = useState(false);

  const sendCampaign = async () => {
    setLoading(true);
    Axios.post(`${hostHeader.url}/api/email/send`, { id }).then((res) => {
      if (res.status === 200) {
        setSuccess(true);
        setLoading(false);
      } else {
        setError(true);
        setLoading(false);
      }
    });
  };

  const buttonToRender = () => {
    switch (status) {
      case "sent":
        return (
          <Link to={`/campaign/${id}`}>
            <button className={"button"}>View Campaign</button>
          </Link>
        );
      case "draft":
        return (
          <button className={"button"} onClick={sendCampaign}>
            Send Now
          </button>
        );
    }
  };
  return (
    <div className={"campaign"}>
      <div className={"top"}>
        <div className={"name"}>{name}</div>
        <div className={"status"} style={{ background: colors[status] }}>
          {status}
        </div>
      </div>
      <div className={"bottom"}>
        <div>
          <div>
            {status === "sent"
              ? `Sent ${sentAt}`
              : status === "scheduled"
              ? `Scheduled at ${scheduledAt}`
              : ""}
          </div>
          <div>{sentTo.length} recipients</div>
        </div>
        <div className={"buttonsContainer"}>{buttonToRender()}</div>
      </div>
    </div>
  );
};

export default memo(Campaign);
