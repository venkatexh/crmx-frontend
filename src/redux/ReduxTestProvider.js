import { Provider } from "react-redux";

const ReduxTestProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

export default ReduxTestProvider;
