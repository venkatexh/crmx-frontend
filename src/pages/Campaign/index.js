import Header from "../../components/Campaign/Header";
import RightNav from "../../components/navigation/RightNav";
import "../../sass/pages/contact.page.scss";
import {useEffect, useState} from "react";
import Axios from "axios";
import hostHeader from "../../config/host";
import Body from "../../components/Campaign/Body";

const Campaign = (props) => {
  const [campaign, setCampaign] = useState({});
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    setLoading(true);
    Axios.get(`${hostHeader.url}/api/campaign/${props.match.params.id}`).then(res => {
      setCampaign(res.data)
      setLoading(false);
    });
  }, []);

  const divToRender = () => {
    return (
      <div className={'contactPage'}>
        {loading ?
          <div className={'contactBodyLoader'}>
            <img
              src={"/loaders/comp_loader.gif"}
              alt={"loader"}
              style={{height: "60px"}}
            />
          </div> :
          <div className={'leftCol'}>
            <Header campaign={campaign}/>
            <Body campaign={campaign}/>
          </div>
        }
        <div className={'rightCol'}>
          <RightNav/>
        </div>
      </div>
    )
  }
  return divToRender();
}

export default Campaign;