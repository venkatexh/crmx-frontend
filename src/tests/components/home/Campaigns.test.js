import { shallow } from "enzyme";
import React from "react";
import Campaign from "../../../components/home/campaigns/Campaign";
import CampaignContainer from "../../../components/home/campaigns/CampaignContainer";
import CampaignHeader from "../../../components/home/campaigns/CampaignHeader";
import ReduxTestProvider from "../../../redux/ReduxTestProvider";

it("should be campaigns container component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <Campaign
          name={"test campaign"}
          status={"sent"}
          sentAt={"today"}
          scheduledAt={"tomorrow"}
          sentTo={["email1@gmail.com", "email2@gmail.com"]}
          id={"testid1234"}
        />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});

it("should be campaigns container component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <CampaignContainer />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});

it("should be campaigns header component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <CampaignHeader />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});
