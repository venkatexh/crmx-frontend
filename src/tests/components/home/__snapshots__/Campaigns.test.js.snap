// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be campaigns container component snapshot 1`] = `ShallowWrapper {}`;

exports[`should be campaigns container component snapshot 2`] = `ShallowWrapper {}`;

exports[`should be campaigns header component snapshot 1`] = `ShallowWrapper {}`;
